File: sdks/ruby/lib/novabilling/payments/types/list_payments_request.rb
# frozen_string_literal: true

module Novabilling
  module Payments
    module Types
      class ListPaymentsRequest < Internal::Types::Model
        field :status, -> { String }, optional: true, nullable: false
        field :provider, -> { String }, optional: true, nullable: false
        field :invoice_id, -> { String }, optional: true, nullable: false, api_name: "invoiceId"
        field :date_from, -> { String }, optional: true, nullable: false, api_name: "dateFrom"
        field :date_to, -> { String }, optional: true, nullable: false, api_name: "dateTo"
        field :page, -> { Integer }, optional: true, nullable: false
        field :limit, -> { Integer }, optional: true, nullable: false
      end
    end
  end
end

File: sdks/php/src/Payments/Requests/ListPaymentsRequest.php
<?php

namespace NovaBilling\Payments\Requests;

use NovaBilling\Core\Json\JsonSerializableType;

class ListPaymentsRequest extends JsonSerializableType
{
    /**
     * @var ?string $status
     */
    public ?string $status;

    /**
     * @var ?string $provider
     */
    public ?string $provider;

    /**
     * @var ?string $invoiceId
     */
    public ?string $invoiceId;

    /**
     * @var ?string $dateFrom
     */
    public ?string $dateFrom;

    /**
     * @var ?string $dateTo
     */
    public ?string $dateTo;

    /**
     * @var ?float $page
     */
    public ?float $page;

    /**
     * @var ?float $limit
     */
    public ?float $limit;

    /**
     * @param array{
     *   status?: ?string,
     *   provider?: ?string,
     *   invoiceId?: ?string,
     *   dateFrom?: ?string,
     *   dateTo?: ?string,
     *   page?: ?float,
     *   limit?: ?float,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->status = $values['status'] ?? null;
        $this->provider = $values['provider'] ?? null;
        $this->invoiceId = $values['invoiceId'] ?? null;
        $this->dateFrom = $values['dateFrom'] ?? null;
        $this->dateTo = $values['dateTo'] ?? null;
        $this->page = $values['page'] ?? null;
        $this->limit = $values['limit'] ?? null;
    }
}
